js/rpg_managers/PluginManager.js

```diff
@@ -51,3 +52,91 @@ PluginManager.loadScript = function(name) {
 PluginManager.onError = function(e) {
     this._errorUrls.push(e.target._url);
 };
+
+PluginManager.createPluginProgram = function(program) {
+    if(this._programCache[program] != void 0) throw new Error('Already initialized pulgin program ' + program);
+    this._programCache[program] = new PluginProgram(program);
+    return this._programCache[program];
+};
+
+PluginManager.runProgram = function(program, params) {
+    if(this._programCache[program] != void 0)
+        this._programCache[program].run(params);
+    else {
+        throw new Error('Failed to find plugin command: ' + program + '.');
+    }
+};
+
+//-----------------------------------------------------------------------------
+// PluginProgram
+//
+// The class that process plugin commands.
+
+
+function PluginProgram() {
+    this.initialize.apply(this, arguments);
+}
+
+PluginProgram.prototype.initialize = function(name) {
+    this._name = name;
+    this._commands = {};
+}
+
+PluginProgram.createParams = function(paramsStyle, args) {
+    if(paramsStyle.length != args.length) throw new Error('Wrong number of arguments (given ' + args.length + ', expected ' + paramsStyle.length + ').')
```

Comment:
This is just a minor readability nitpick, but this line seems needlessly wrong for no good reason. I would recommend writing this as four lines:

```javascript
if (paramsStyle.length != args.length) {
    throw new Error('Wrong number of arguments (given ' + args.length +
        ', expected ' + paramsStyle.length + ').');
}
```